.github/workflows/pypi_publish.yml

```diff
@@ -12,26 +13,33 @@ on:
 
 jobs:
   build-sdist:
-    name: Build package
+    name: Build sdist
```

Comment:
There's a bug in the sdist package version. This is what I find in the CI artifact:
`versioned_hdf5-0.1.dev1+g5317d17.tar.gz`